documentation/configuration.markdown

```diff
@@ -106,7 +106,15 @@ puppet-admin: {
 
 ### `master.conf`
 
-This file contains settings for the Puppet master functionality of Puppet Server. If `allow-header-cert-info` is set to 'true', this allows the `ssl_client_header` and `ssl_client_verify_header` options in puppet.conf to work. By default, this is set to 'false'.
+This file contains settings for Puppet master features, such as node identification and authorization.
+
+In a default installation, this file doesn't exist. You'll need to create it if you want to set non-default values for these settings.
+
+* `allow-header-cert-info` determines whether Puppet Server should use authorization info from the `X-Client_Verify`, `X-Client-CN`, and `X-Client-Cert` HTTP headers. Defaults to `false`.
+
+    This setting is used to enable [external SSL termination.](./external_ssl_termination.markdown) If enabled, Puppet Server will ignore the actual certificate presented to the Jetty webserver, and will rely completely on header data to authorize requests. This is very dangerous unless you've secured your network to prevent any untrusted access to Puppet Server.
```

Comment:
Should we somehow mention that when using HTTPS the certificate will be ignored, but generally HTTP is used between the external terminator and Puppet Server and there is no cert to speak of.